src/sprightly.ts

```diff
@@ -116,7 +124,7 @@ function sprightlyAsync(
     try {
       resolve(sprightly(entryPoint, data, options))
     } catch (error) {
-      reject((error as Error).message)
+      reject((error as SprightlyError).message)
```

Comment:
Throw the error, not the message. This way the user can get the message and other info as well.